index.html

```diff
@@ -4292,6 +4306,454 @@ <h3>Proof Formats</h3>
+the statements in Section <a href="#integrity-of-related-resources"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `refreshService` property conforms to
+the statements in Section <a href="#refreshing"></a>.
+                  </li>
+                  <li>
+If present, ensure that the `evidence` property conforms to
+the statements in Section <a href="#evidence"></a>.
+                  </li>
+                </ol>
+              </li>
+            </ol>
+          </li>
+          <li>
+If no errors were detected to this point, set
+<var>dataModelVerified</var>.<var>document</var> to <var>inputDocument</var>,
+<var>dataModelVerified</var>.<var>verified</var> to `true`.
+          </li>
+          <li>
+Return <var>dataModelVerified</var>.
+          </li>
+        </ol>
+
+      </section>
+
+      <section class="normative">
+        <h3>Verify Securing Mechanism</h3>
+
+        <p>
+The algorithm in this section can be used to verify whether a securing mechanism
+that is associated with a sequence of bytes is valid. This algorithm takes a
+byte sequence (bytes <var>inputBytes</var>) and a media type
+(string <var>inputMediaType</var>) as inputs, and produces an object
+(<var>securingMechanismVerified</var>) that contains the following:
+        </p>
+
+        <ul>
+          <li>
+a verification status (boolean <var>status</var>)
+          </li>
+          <li>
+a verified document (object <var>document</var>)
+          </li>
+          <li>
+a media type (string <var>mediaType</var>)
+          </li>
+          <li>
+zero or more warnings (array of <a>ProblemDetails</a> <var>warnings</var>)
+          </li>
+          <li>
+zero or more errors (array of <a>ProblemDetails</a> <var>errors</var>)
+          </li>
+        </ul>
+
+        <p>
+The securing mechanism verification algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set <var>securingMechanismVerified</var> to an empty object and initialize the
+following fields: <var>status</var> to `false`, <var>warnings</var> to an
+empty array, and <var>errors</var> to an empty array.
+          </li>
+          <li>
+Determine the specific mechanism for extracting the <a>conforming document</a>
+(<var>extractDocument</var>):
+            <ol class="algorithm">
+              <li>
+If the <var>inputMediaType</var> is `application/vc+ld+json` or
+`application/vp+ld+json`, the <var>extractDocument</var> function takes
+<var>inputBytes</var> as an input parameter, parses the value as JSON,
+and returns the result as an object. If JSON parsing fails, a
+<a href="#PARSING_ERROR">PARSING_ERROR</a> is appended to
+<var>securingMechanismVerified</var>.<var>errors</var>.
+              </li>
+              <li>
+If the <var>inputMediaType</var> is associated with the JOSE or COSE
+specifications, the [[?VC-JOSE-COSE]] specification defines a process for
+extracting the <a>conforming document</a>.
```

Comment:
Added the issue marker in 90a4c748f723c7aa9cf8664f702d6ffd9829b3aa.

Reply:
There won't be an extract in JOSE or COSE, because that securing mechanism just signs & verifies bytes.... which is better than what is being contemplated in this PR.

Reply:
@msporny As you say earlier, `data integrity` tries to validate untrusted data before verifying, and then returns a verification result.

`vc-jose-cose` tries to validate enough to finish key discovery, verify, and then validate the result is a conforming document.

regardless of the order, both securing mechanisms yield the following booleans:

- verified (signature check)
- validated (schema check)
- verified payload (conforming document)

The part of validating a "verified conforming document" can be shared, modulo my comments about proof.

The signature and schema checks cannot be shared, they are securing specification media type specific, but this generic normative algorithm can assert that there be a boolean response from "Verify Securing Mechanism" 

I don't think implementers of COSE Sign1 or SD-JWT should have to read about `extractDocument` or any other data integrity specific details, but those details can easily be preserved in the data integrity spec, and the outcome can be the same.

For example, lets consider an SD-JWT VP.

The SD-JWT verification process produces a "conformant document".

The media type for SD-JWT is currently `application/vp+ld+json+sd-jwt`

The conformant document (verified payload) media type is `application/vp+ld+json`

This document might have a `proof` or it might not, and each embedded credential might have a `proof` or it might not.

The details of this proof property are already addressed in the securing mechanism, it treats the entire SD-JWT payload as JSON, `proof` is just another JSON member to SD-JWT.

This also consistent with how C2PA treats proof:

> W3C Verifiable Credentials are used in this specification to decorate the actors identified in assertions with more information, potentially providing additional trust signals. Although these W3C Verifiable Credentials can include proofs of their own authenticity, they are not a mechanism for verifying that a particular actor authorised a claim, assertion or piece of metadata. Any validation or usage of the W3C Verifiable Credential is out of scope of this specification and has no bearing on the [C2PA Trust Model](https://c2pa.org/specifications/specifications/1.3/specs/C2PA_Specification.html#_trust_model).


In data integrity, depending on proof chaining or proof sets, the result of the verify operation is a conforming document, and some proof details (verification checks for proof set or proof chain, see https://w3c.github.io/vc-data-integrity/#verify-proof-sets-and-chains)

In the context of the core data model verification algorithm, the only thing that matters here is "did verification fail"... as in "any proof verified : false".

If that happens, we don't need to do any more validation, and we certainly don't need to comment on `<var>extractDocument</var>`, in much the same way that we don't need to talk about: https://w3c.github.io/vc-data-integrity/#verify-proof

in the core data model.

Reply:
@jyasskin wrote:
> that the securing method should return the secured document from the same call that checks whether the signatures were valid.

That's what is being attempted in this algorithm. What lead you to believe we're doing anything other than that?

Reply:
I think both @OR13 and I would like `verifyProof` to return the document that becomes <var>securingMechanismVerified</var>.<var>document</var> (only if cryptographic verification succeeds), instead of having a separate `extractDocument` operation that does it. Once that's the interface, Orie's removal of the "Other extracting functions are possible" section is fine, since there aren't any dedicated extracting functions.